File: examples/bad-code.py
# Python example with various issues

import os
import subprocess

# Security issues
password = "admin123"  # Hardcoded password
api_key = "sk-1234567890"  # Hardcoded API key

def process_data(user_input):
    # Very dangerous - command injection vulnerability
    result = subprocess.call(user_input, shell=True)
    
    # Multiple nested conditions (high complexity)
    if result == 0:
        if user_input and len(user_input) > 0:
            if user_input.startswith("valid"):
                if user_input.endswith("data"):
                    if "special" in user_input:
                        print("Success")
                        print("Step 1")
                        print("Step 2")
                        print("Step 3")
                        print("Step 4")
                        print("Step 5")
                        print("Step 6")
                        print("Step 7")
    
    # TODO: Fix this security issue
    # FIXME: Refactor this function
    
    # Magic numbers
    value = user_input * 42 + 137 - 256
    another = value / 999 + 12345
    
    return value

def very_long_function_name_that_exceeds_recommended_line_length_and_should_be_refactored():
    return "This function name is way too long"

# No error handling
def risky_operation(filename):
    file = open(filename, 'r')
    data = file.read()
    # File never closed - resource leak
    return data

File: examples/bad-code.js
// Sample JavaScript file with various code quality issues

function processUserData(userData) {
  // This is a long function that does too many things
  const password = "hardcoded123"; // Security issue!
  const apiKey = "sk-1234567890abcdef"; // Another security issue!
  
  console.log("Processing user:", userData);
  console.log("Step 1");
  console.log("Step 2");
  console.log("Step 3");
  console.log("Step 4");
  console.log("Step 5");
  console.log("Step 6");
  
  // TODO: Fix this later
  // FIXME: This is broken
  
  if (userData && userData.name && userData.email && userData.age && userData.address && userData.phone) {
    if (userData.age > 18 && userData.age < 65 && userData.verified === true || userData.premium === true) {
      if (userData.country === "US" || userData.country === "UK" || userData.country === "CA") {
        document.getElementById("user-info").innerHTML = userData.name; // XSS vulnerability!
        
        // Magic numbers everywhere
        const result = userData.score * 42 + 137 - 256;
        const another = result / 999 + 12345;
        
        try {
          eval(userData.customCode); // Very dangerous!
        } catch (e) {
          // Empty catch block
        }
        
        for (let i = 0; i < 100; i++) {
          if (i % 2 === 0) {
            if (i % 3 === 0) {
              if (i % 5 === 0) {
                console.log(i);
              }
            }
          }
        }
        
        return true;
      }
    }
  }
  return false;
}

function anotherVeryLongLineOfCodeThatExceedsTheRecommendedLineLengthAndShouldBeRefactoredIntoSmallerPiecesForBetterReadability() {
  return "This line is way too long and violates coding standards";
}

const user = {
  name: "John",
  password: "admin123"
};
